python/ray/util/spark/start_ray_node.py

```diff
@@ -41,14 +41,14 @@
         if arg.startswith(temp_dir_arg_prefix):
             temp_dir = arg[len(temp_dir_arg_prefix) :]
 
-    if temp_dir is None:
-        raise ValueError("Please explicitly set --temp-dir option.")
-
-    temp_dir = os.path.normpath(temp_dir)
+    if temp_dir is not None:
+        temp_dir = os.path.normpath(temp_dir)
+    else:
+        # This case is for global mode Ray on spark cluster
+        lock_file = "/tmp/ray_on_spark_tmp.lock"
```

Comment:
Why is lock_file path '/tmp/..' instead of `start_hook.get_default_temp_root_dir()`

Reply:
Also we don't set temp_dir in global mode, then we can't clean up the temp dir, below `try_clean_temp_dir_at_exit` refers to temp_dir variable directly.